.github/workflows/update-version-test.yml

```diff
@@ -0,0 +1,41 @@
+name: Update Version
+
+on:
+  workflow_call:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    env:
+      SHARED: /tmp/workdir/pki
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Setup git
+        run: |
+          sudo apt-get -y install git
+          git config --global user.name "Dr. John Doe"
+          git config --global user.email jdoe@example.com
+
+      - name: Update to version with a phase
+        run: |
+          ./update_version.sh 12 1 1 beta1
+          git tag --points-at HEAD  > actual
+          echo v12.1.1.beta1 > expected
+          diff expected actual
+
+      - name: Update to version without a phase
+        run: |
+          ./update_version.sh 12 1 2
+          git tag --points-at HEAD  > actual
+          echo v12.1.2 > expected
+          diff expected actual
+
+      - name: Update to version with a phase again
+        run: |
+          ./update_version.sh 12 1 3 alpha1
```

Comment:
We can use alpha/beta again in the next minor version before GA, i.e. `12.2.0-alpha1`.